crates/core/src/operations/update.rs

```diff
@@ -1005,10 +1005,10 @@ pub(crate) async fn request_update(
             .closest_potentially_caching(&key, [sender.peer().clone()].as_slice());
 
         if let Some(target) = remote_target {
-            // Subscribe to the contract
+            // Subscribe to the contract - sender is ourselves, no upstream_addr needed
```

Comment:
"sender is opurselves" but this comes from "remote_target" ? Soemthing is off here.